PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/ControlScript/MasterControl/TouchJump.rbxmx

```diff
@@ -97,6 +100,18 @@ function TouchJump:Create(parentFrame)
 		end
 	end)
 	
+	if humanoid then
+		
+		humanoid.Changed:connect(function(prop)
```

Comment:
I think for this we would want to store the connection and disconnect on disable, since humanoid can change. We should probably also listen to player humanoid changing? would make this more robust + change in MasterControl unnecessary.